### PR TITLE
refactor: partial migration to typescript

### PR DESCRIPTION
**Pull request by santimirandarp**
Summary:

* Perform the partial TS migration (uses js files.)
* Vitest (tests were taking too long to run locally otherwise)
* Remove Rollup entirely
* Refactors (only for of loops)
* Small fixes
* Updated/removed dependencies.

To check:

* Why there is such a large diff (replaced snapshots ?)

### CODE REVIEW

**Comment by lpatiny** [comment]
There are some issues with jsDoc because in tsDoc there should not be the type

```js
/**
 * Returns the hose code for specific atom numbers
 * @param {import('openchemlib').Molecule} molecule - The OCL molecule with expandedImplicitHydrogens and ensureHeterotopicChiralBonds
 * @param {object} [options={}]
 * @param {string[]} [options.allowedCustomLabels] Array of the custom labels that are considered as root atoms. By default all atoms having a customLabel
 * @param {number} [options.minSphereSize=0] Smallest hose code sphere
 * @param {number} [options.maxSphereSize=4] Largest hose code sphere
 * @param {number} [options.kind=FULL_HOSE_CODE] Kind of hose code, default usual sphere
 */
```

Should become something like:

/**
 * Create an array with numbers starting from "from" with step "step" of length "length"
 *
 * @param options - options
 * @return - array of distributed numbers with step "step" from "from"
 */
```

options should also become a specific TS type but this could be done after.

**Comment by santimirandarp** [comment]
There is still an error in the tests @lpatiny

File: src/reaction/utils/applyOneReactantReaction.js [hunk 1]
```diff
@@ -56,23 +53,17 @@ export function applyOneReactantReaction(reactants, reactions, options) {
...
+                });
               });
             }
           }
           // if there is at least one product, we add the reaction to the results
           if (products.length > 0) {
-            // eslint-disable-next-line no-unused-vars
-            const { oclReaction, needToBeCharged, ...reactionWithoutOCL } =
-              reaction;
+            const { oclReaction, ...reactionWithoutOCL } = reaction;
```
Comment (lpatiny) [inline src/reaction/utils/applyOneReactantReaction.js:66]: We have an eslint-error but I don't understand why. It seems to me that @targos you allowed this explicitely in https://github.com/cheminfo/eslint-config-cheminfo-typescript/commit/5c77912979c6d06aff850ffc86433114d3461914 (`ignoreRestSiblings: true`)

Comment (targos) [reply src/reaction/utils/applyOneReactantReaction.js:66]: TypeScript rules are only applied to TypeScript files.

**Pull request merged**